### DISCUSSION TITLE
Redirect value does not equal option under "has"?

### DISCUSSION THREAD

**Discussion by Wtrapp** [vercel/next.js#24166] [Help]
Is there a way to match if the has value does not equal a value?

```
async redirects() {
    return [
        {
          source: '/secretpage',
          has: [
            {
              type: 'header',
              key: 'host',
              value: '!=wrongsite.com',
            },
          ],
          permanent: false,
          destination: 'https://goodsite.com/',
        },
      ]
  }
```

**Comment by ezeartigas** [comment]
I'm looking for the same. I need to verify if the "token" cookie doesn't exist and redirect to /login

**Comment by laugharn** [comment]
@Wtrapp To negatively match against a value in a header you could do something like `^(?!wrongsite.com$).*$`

@ezeartigas I'm working on the same thing and came here to put in a feature request, because I'm _pretty sure_ you can't match against something being undefined.
